### PR TITLE
Configuration move to semver -centric version checks for queries.

### LINKED ISSUE #326: Enhance rest url maintenance
Move to semver library to allow embedding rule comparison in configuration.
Consolidate retry and file extension attributes into configuration.

### PR DESCRIPTION
**Pull request by nemonster**
Dependency and version updates.
Minor mods to logging config.
Separated out rest calls to elastic and logstash specific and changed versioning format from release based to call based.
Added test to validate format of rest configuration files at build time.
New classes for refactored rest entries.
Closes #326 , #277

### CODE REVIEW

File: src/main/java/com/elastic/support/rest/RestEntryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.swing.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class RestEntryBuilder {
```
Comment (pickypg) [inline src/main/java/com/elastic/support/rest/RestEntryBuilder.java]: Nitpick: This is more of a `Factory` than a `Builder`. It also doesn't seem to be used, but I'm guessing that's for the next PR?

File: src/main/resources/logstash-rest.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+logstash_version:
+  extension: ".json"
+  versions:
+    "> 0.0.0": "/?pretty"
+
+logstash_node:
+  extension: ".json"
+  versions:
+    "> 0.0.0": "/_node?pretty"
```
Comment (pickypg) [inline src/main/resources/logstash-rest.yml]: These versions aren't true, right? I forget what version we actually added APIs to Logstash, but I didn't think it was until at least 2.x? Obviously with the `version` check API we can't know until we try.

File: src/main/java/com/elastic/support/rest/RestEntryBuilder.java (hunk 1 above)
Comment (nemonster) [reply src/main/java/com/elastic/support/rest/RestEntryBuilder.java]: True, it changed substantially at one point and I never renamed it. Next PR will have the implementation. At some point I'll probably rename it but I had other higher priorities.

File: src/main/resources/logstash-rest.yml (hunk 2 above)
Comment (nemonster) [reply src/main/resources/logstash-rest.yml]: It was 5.x, but if you look, the current version has no checks in code, just instructions not to use is before v.5. So I kept it consistent with what was there for now. The fact that nobody has ever raised an issue on that probably means it never got tried on an earlier version, At some point I'll make another pass through to see if there's any differentiation that requires a version check?

**Pull request merged**